### DISCUSSION TITLE
Updating to Cockpit 329 removes cockpit-pcp

### DISCUSSION THREAD

**Discussion by compbl** [cockpit-project/cockpit#21343] [General]
When I attempt to update to Cockpit 3.29 via the software updates page I am receiving an error "The following packages have unmet dependencies:  cockpit-bridge: Conflicts: cockpit-pcp"  I am assuming I have to remove cockpit-pcp, but cannot find how to do that.   Any help would be appceiated.   Running on Ubuntu 22.04.5 LTS

Thanks

**Comment by martinpitt** [comment]
Right, the cockpit-pcp package is obsolete, and should be removed on upgrades. See https://cockpit-project.org/blog/cockpit-326.html . How exactly did you try to upgrade, and what did it say?

**Reply by compbl** [reply]
I am doing the upgrade from within Cockpit itself. 

![image](https://github.com/user-attachments/assets/8e8a05e6-0c09-4bfb-bfb4-d846f629b67c)

![image](https://github.com/user-attachments/assets/61f49e50-2e46-41d9-b06b-46348939fb98)

**Comment by martinpitt** [comment]
Ah, thanks @compbl ! I'm starting from the basics, booted a clean 22.04 machine with some cockpit packages installed:

```
# dpkg -l | grep ^ii.*cockpit
ii  cockpit                                          264-1ubuntu0.22.04.1                    all          Web Console for Linux servers
ii  cockpit-bridge                                   264-1ubuntu0.22.04.1                    amd64        Cockpit bridge server-side component
ii  cockpit-pcp                                      264-1ubuntu0.22.04.1                    amd64        Cockpit PCP integration
ii  cockpit-system                                   264-1ubuntu0.22.04.1                    all          Cockpit admin interface for a system
ii  cockpit-ws             
```

and apt itself can do it fine:
```
 apt upgrade -t jammy-backports cockpit
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
Get more security updates through Ubuntu Pro with 'esm-apps' enabled:
  redis-server redis-tools redis
Learn more about Ubuntu Pro at https://ubuntu.com/pro
The following packages will be REMOVED:
  cockpit-pcp
The following packages will be upgraded:
  cockpit cockpit-bridge cockpit-system cockpit-ws dmidecode libdebuginfod-common libdebuginfod1 libdw1 libelf1 libsoup-gnome2.4-1
  libsoup2.4-1 libsoup2.4-common needrestart
```

Setting up your scenario (older backport) is a bit more effort, I'll do that later.

**Reply by compbl** [reply]
Is it suggestable to do the _apt upgrade -t jammy-backports cockpit_ ?

**Reply by martinpitt** [reply]
That'll circumvent the bug and update it on the command line, so that will work for you.

**Reply by compbl** [reply]
Worked... Thank You!

**Comment by martinpitt** [comment]
Moved to #21344 , as that is an actual bug.
